File: OOP/oopA4.py
import random

class BigCat:
   def __init__(self):
      speed=5
      strength=5
      intelligence=5
      health=5
      durability=5

class Lion(BigCat):
    def __init__(self):
      super().__init__()
      self.strength = 50
      self.health = 50
    
    def king(self,bigcat):
        if isinstance(bigcat,Cheetah):
          random_number = random.randint(1,100)

          if random_number <= 60:
            print("Cheetah has escaped")
            return 

        speed=0
        strength=0
        intelligence=0
        health=0
        durability=0

# lion = Lion()
# lion2 = Lion()

# lion.king()

File: ES2015+/OOPA4.js
class BigCat{
    constructor(){
        this.speed = 5 
        this.strength = 5 
        this.intelligence = 5 
        this.health = 5 
        this.durability = 5
    }
}

class Lion extends BigCat{
    constructor(){
        super()
        this.health = 50
        this.strength = 50
    }

    king(bigcat){
        if(bigcat instanceof Cheetah){
            let randomNumber = Math.round( Math.random() * 100 ) // generate random number from 0 to 100

            if(randomNumber <= 60){ //Cheetah has 60% chance of leaving unscathed
                console.log("The cheetah escaped unscathed")
                return
            }
        }

        bigcat.speed = 0 
        bigcat.strength = 0 
        bigcat.intelligence = 0 
        bigcat.health = 0 
        bigcat.durability = 0
    }
}

class Cheetah extends BigCat{

}

class Leopard extends BigCat{

}